.github/workflows/frontend-unit-tests.yml

```diff
@@ -2,16 +2,27 @@ name: Frontend Tests
 
 on:
   push:
+    branches:
+      - main
     paths:
       - 'frontend/**'
       - '!frontend/testing/cypress/**'
       - '!frontend/stats/**'
       - 'testdata/**'
       - '.github/workflows/frontend-unit-tests.yml'
       - 'package.json'
+  pull_request:
+    paths:
+      - 'frontend/**'
+      - '!frontend/testing/cypress/**'
+      - '!frontend/stats/**'
+      - 'testdata/**'
+      - '.github/workflows/frontend-unit-tests.yml'
+      - 'package.json'
+  workflow_dispatch:
```

Comment:
Is this added intentionally, or is it included for testing purposes to allow triggering the pipeline on demand?

Reply:
Yes, this was added intentionally. I have read that is some cases, the `changedSince` flag do not detect changes, so I wanted to give the user the option to manually run the tests for the entire solution in its PR without needing to merge it into main.

Reply:
Ok, this should be informed to the team, otherwise our fellow team members may think there are no tests to run. But this is also a backdoor for failing tests, right? If our f.ex juniors don't know they should manually trigger the tests in some cases, we might have a failing pipeline within the RP that is not detected. 😊

Reply:
Yes, I agree, but the option to run the tests manually was intended for very rare cases, such as when you change the test setups, so it is not something that should be used frequently.
I'm not entirely sure how the `changedSince` flag will perform, so the idea here is to evaluate it over time. In the best-case scenario, the flag catches all our changes, and maybe once every six months, the tests fail when merging to main. In that case, we could consider keeping this flag as it prevents running 4000 tests on each push. However, if we see that Jest frequently misses tests, then we should definitely not use it.

Reply:
To summarize the pros and cons of using the flag:
- Pros: we avoid running 4000 tests on each push, saving time and resources
- Cons: in rare cases, Jest might not detect the changes. The solution will still be fully tested during the merge to the main branch, alerting us if issues occur, but this means risking merging untested code into the main branch, which is not optimal.

One option is to give it a try and identify scenarios where the changes are not detected, to see if they are indeed rare or not. Or to not use the flag at all to prevent risking merging untested code. But in this case, the test time per push would go back from 1 minute to 11 minutes.

Reply:
Ah ok, I get it 😊 Yes, agreed, I'll set up a short meeting/demo to explain this 👍